app/services/email_service/ses/client.rb

```diff
@@ -38,17 +49,52 @@ def verify_address(email: )
         raise ArgumentError.new("Missing mandatory value for email parameter.")
       end
 
-      begin
-        response = @ses.verify_email_identity(email_address: email)
-        if response.successful?
-          Result::Success.new()
-        else
-          Result::Error.new(response.error)
+      log_request_response(:verify_email_identity, {email_address: email}) do
+        begin
+          response = @ses.verify_email_identity(email_address: email)
+          if response.successful?
+            Result::Success.new()
+          else
+            Result::Error.new(response.error)
+          end
+        rescue StandardError => e
+          Result::Error.new(e)
         end
-      rescue StandardError => e
-        Result::Error.new(e)
       end
     end
 
+    def set_notification_topic(email:, type:)
+      nt = DataTypes::NotificationTopic.call({email: email, type: type})
+      method_params = {identity: nt[:email],
+                       notification_type: nt[:type].to_s.capitalize,
+                       sns_topic: @sns_topic}
```

Comment:
initialize method validates that it exists so the only way to mess this up is to mess with instance variables after construction. I might still add a validation because that's a valid call (nil sns topic) and it clears the sns topics from the email.